gson/src/main/java/com/google/gson/reflect/ReflectionAccessor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2017 The Gson authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gson.reflect;
+
+import com.google.gson.reflect.impl.PreJava9ReflectionAccessor;
+import com.google.gson.reflect.impl.UnsafeReflectionAccessor;
+import com.google.gson.util.VersionUtils;
+
+import java.lang.reflect.AccessibleObject;
+
+/**
+ * Provides a replacement for {@link AccessibleObject#setAccessible(boolean)}, which may be used to
+ * avoid reflective access issues appeared in Java 9, like {@link java.lang.reflect.InaccessibleObjectException}
+ * thrown or warnings like
+ * <pre>
+ *   WARNING: An illegal reflective access operation has occurred
+ *   WARNING: Illegal reflective access by ...
+ * </pre>
+ * <p/>
+ * Works both for Java 9 and earlier Java versions.
+ */
+public abstract class ReflectionAccessor {
```

Comment:
package-private constructor please

Reply:
Why adding a constructor here? The class is abstract, so the default (public) constructor still cannot be used by anyone except of implementing classes

Reply:
Because it prevents subclasses outside the package.

Reply:
Yes, but why shall we prevent such subclasses? I am not against it, but just do not understand...